sdks/java/io/rabbitmq/src/test/java/org/apache/beam/sdk/io/rabbitmq/RabbitMqIOTest.java

```diff
@@ -168,13 +209,13 @@ public void testReadExchange() throws Exception {
                 } catch (Exception e) {
                   LOG.error(e.getMessage(), e);
                 }
-                for (int i = 0; i < maxNumRecords; i++) {
+                for (int i = 0; i < testPlan.getNumRecordsToPublish(); i++) {
                   try {
                     finalChannel.basicPublish(
-                        "READEXCHANGE",
-                        "test",
+                        exchange,
+                        testPlan.publishRoutingKeyGen().get(),
```

Comment:
Exchange is now taken from the Read definition rather than being hard-coded. The routing key is taken from the Plan's routing key generator.